lua/cmake-tools/init.lua

```diff
@@ -20,1075 +19,1018 @@ local full_cmd = ""
 
 --- Setup cmake-tools
 function cmake.setup(values)
-  const = vim.tbl_deep_extend("force", const, values)
-  config = Config:new(const)
-  -- preload the autocmd if the following option is true. only saves cmakelists.txt files
-  if const.cmake_regenerate_on_save then
-    cmake.create_regenerate_on_save_autocmd()
-  end
-
-  -- auto reload previous session
-  if cmake.is_cmake_project() then
-    local old_config = _session.load()
-    if next(old_config) ~= nil then
-      config:update_build_dir(old_config.build_directory)
-      config.generate_options = old_config.generate_options
-      config.build_options = old_config.build_options
-
-      config.build_type = old_config.build_type
-      config.build_target = old_config.build_target
-      config.launch_target = old_config.launch_target
-      config.launch_args = old_config.launch_args
-      config.kit = old_config.kit
-      config.configure_preset = old_config.configure_preset
-      config.build_preset = old_config.build_preset
-    end
-  end
+	if values.cmake_executor ~= nil then
+		const.cmake_executor = values.cmake_executor
+	end
```

Comment:
Why should we handle `cmake_executor` specifically?

Reply:
Changed, but it still does have a specific way to handle it, because using deepcopy with it will mix options from multiple executors